### PR TITLE
feat: add Docker sbom spec

### LINKED ISSUE #1117: docker: add sbom spec
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [ ] I would like to work on this.

### What CLI tool does this relate to?

docker

### Which statement makes the most sense?

This completion spec isn't complete

### Issue Details

Add `docker sbom` support 

https://www.docker.com/blog/announcing-docker-sbom-a-step-towards-more-visibility-into-docker-images/?utm_campaign=2022-04-07-sbomblog&utm_medium=social&utm_source=twitter

### PR DESCRIPTION
**Pull request by dirien**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Fixes #1117 

**What is the current behavior? (You can also link to an open issue here)**
No sbom spec
**What is the new behavior (if this is a feature change)?**
`docker sbom` support
**Additional info:**

image

### CODE REVIEW

**Review by mschrage — APPROVED** [review]

File: src/docker.ts [hunk 1]
```diff
@@ -1985,6 +1998,81 @@ default-cgroupns-mode option on the daemon (default)`,
...
+        description:
+          "File to write the default report output to (default is STDOUT)",
+        args: {
+          name: "file",
+        },
+      },
+      {
+        name: "--platform",
+        description:
+          "An optional platform specifier for container image sources (e.g. 'linux/arm64', 'linux/arm64/v8', 'arm64', 'linux')",
```
Comment (fedeci) [inline src/docker.ts:2061]: Can you add a suggestions object to this arg containing the currently available platforms?
```ts
args: {
	name: "platform",
	suggestions: [
		"linux/arm64",
		"linux/arm64/v8",
		"arm64",
		// ...other targets
	]
}
```

File: src/docker.ts [hunk 2]
```diff
@@ -1985,6 +1998,81 @@ default-cgroupns-mode option on the daemon (default)`,
...
+      {
+        description: "Exclude paths from being scanned using a glob expression",
+        name: "--exclude",
+        args: {
+          name: "paths",
+        },
+      },
+      {
+        description:
+          'Report output format, options=[syft-json cyclonedx-xml cyclonedx-json github-0-json spdx-tag-value spdx-json table text] (default "table")',
```
Comment (fedeci) [inline src/docker.ts:2023]:
```suggestion
          'Report output format',
```

File: src/docker.ts [hunk 3]
```diff
@@ -1985,6 +1998,81 @@ default-cgroupns-mode option on the daemon (default)`,
...
+            "spdx-json",
+            "table",
+            "text",
+          ],
+          default: "table",
+        },
+      },
+      {
+        description:
+          '[experimental] selection of layers to catalog, options=[squashed all] (default "squashed")',
```
Comment (fedeci) [inline src/docker.ts:2042]:
```suggestion
          '[experimental] selection of layers to catalog',
```

File: src/docker.ts [hunk 4]
```diff
@@ -50,6 +50,19 @@ const dockerGenerators: Record<string, Fig.Generator> = {
...
     },
   },
+  allLocalImagesWithRepository: {
+    script: `docker image ls --format '{{ json . }}'`,
+    postProcess: function (out) {
+      return out
+        .split("\n")
+        .map((line) => JSON.parse(line))
+        .map((i) => ({
+          name: `${i.Repository}`,
```
Comment (fedeci) [inline src/docker.ts:60]: nit:
```suggestion
          name: i.Repository,
```

File: src/docker.ts [hunk 5]
```diff
@@ -1985,6 +1998,81 @@ default-cgroupns-mode option on the daemon (default)`,
...
+          suggestions: ["squashed", "all"],
+          default: "squashed",
+        },
+      },
+      {
+        name: ["-o", "--output"],
+        description:
+          "File to write the default report output to (default is STDOUT)",
+        args: {
+          name: "file",
```
Comment (fedeci) [inline src/docker.ts:2055]:
```suggestion
          name: "file",
          template: ["filepaths"],
          suggestCurrentToken: true,
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks @dirien, I left some comments.

File: src/docker.ts (hunk 5 above)
Comment (dirien) [reply src/docker.ts:2053]: thx for the feedback -> done :)

File: src/docker.ts (hunk 4 above)
Comment (dirien) [reply src/docker.ts]: thx for the feedback -> done :)

File: src/docker.ts (hunk 3 above)
Comment (dirien) [reply src/docker.ts]: thx for the feedback -> done :)

File: src/docker.ts (hunk 2 above)
Comment (dirien) [reply src/docker.ts]: thx for the feedback -> done :)

**Comment by dirien** [comment]
hi @fedeci, thanks for reviewing the PR. Did you suggested changes. :)

**Pull request merged**